docs/guillotine.adoc

```diff
@@ -0,0 +1,1349 @@
+
+
+{zwsp} +
+{zwsp} +
+{zwsp} +
+
+
+== Groundwork: movie items
+
+This first stage should be easy enough, almost entirely repeating steps you've been through in previous chapters. We'll make a _movie_ content type, set up react4xp to preview-render it with react components (but with <<#controller_mapping, a little twist>>), and add some movie items that will be listed when the site is done.
+
+[NOTE]
+====
+This entire chapter builds on the <<imports-and-dependency-chunks#webpack_config, config setup from the previous lesson>>: _react4xp.properties_, _webpack.config.react4xp.js_ and the extra NPM packages should be set up like that.
+
+If you haven't completed that section already, better take a couple of minutes and do that before proceeding.
+====
+
+{zwsp} +
+
+=== Movie content type
+
+When the setup is ready, we'll start by adding a _movie_ *content type*, with an ImageSelector for a poster `image`, a simple HtmlArea with a movie `description`, a numeral Long field for adding the release `year` and an array of `actor` names:
+
+.site/content-types/movie/movie.xml:
+[source,xml,options="nowrap"]
+----
+<content-type xmlns="urn:enonic:xp:model:1.0">
+  <display-name>Movie</display-name>
+  <description>Moving images often reflecting culture</description>
+  <super-type>base:structured</super-type>
+
+  <form>
+    <input name="image" type="ImageSelector">
+        <label>Movie poster</label>
+        <occurrences minimum="1" maximum="1"/>
+    </input>
+
+    <input name="description" type="HtmlArea">
+        <label>Description</label>
+        <config>
+            <exclude>*</exclude>
+            <include>Bold Italic Underline</include>
+        </config>
+    </input>
+
+    <input name="year" type="Long">
+        <label>Release year</label>
+        <occurrences minimum="1" maximum="1"/>
+    </input>
+
+    <input name="actor" type="TextLine">
+        <label>Actor</label>
+        <occurrences minimum="0" maximum="0"/>
+    </input>
+  </form>
+</content-type>
+----
+
+{zwsp} +
+
+=== React components
+
+Next, we'll set up a few react components for visualizing each movie item.
+
+The *entry*, _Movie.jsx_, will take care of rendering a preview of each movie content item in content studio later:
+
+.react4xp/myEntries/Movie.jsx:
+[source,javascript,options="nowrap"]
+----
+import React from 'react'
+
+import Movie from '../shared/Movie';
+
+export default (props) => <Movie {...props} />;
+----
+
+This is a pure entry wrapper that just imports the next react component from _react4xp/shared_.
+
+Why import code from _shared_ instead of keeping it all in the entry? Two reasons:
```

Comment:
Hmm, how about this: 

```
Why import code from _shared_ instead of keeping it all in the entry? Firstly, it's a good rule of thumb to keep entries slim, for better optimization. And secondly, in addition to a Content Studio preview for single movies, we're going to use the *imported components* in the actual movie list too, for each single movie in the list.
```

Reply:
✅